pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -368,9 +368,13 @@ cdef class _Timestamp(ABCTimestamp):
                 self = type(other)(self)
 
             # validate tz's
-            if not tz_compare(self.tzinfo, other.tzinfo):
-                raise TypeError("Timestamp subtraction must have the "
-                                "same timezones or no timezones")
+            naive_timezones = [
+                el.tzinfo is None or el.tzinfo.utcoffset(el) is None
+                for el in (self, other)
+            ]
+            if any(naive_timezones) and not all(naive_timezones):
+                raise TypeError("Cannot compute time difference between"
+                                " timezone-aware and naive datetime values.")
```

Comment:
nitpick: put the space at the end of the previous line instead of the beginning of this line

Reply:
yeah, the CI suggested this as well